rotkehlchen/exchanges/bitcoinde.py

```diff
@@ -0,0 +1,249 @@
+        self.session.headers.update({
+            'x-api-signature': signature,
+        })
+        return signature
+
+    def _api_query(
+            self,
+            verb: str,
+            path: str,
+            options: Optional[Dict] = None,
+    ) -> Dict:
+        """
+        Queries Bitcoin.de with the given verb for the given path and options
+        """
+        assert verb in ('get', 'post', 'push'), (
+            'Given verb {} is not a valid HTTP verb'.format(verb)
+        )
+
+        request_path_no_args = '/v4/' + path
+
+        data = ''
+        if not options:
+            request_path = request_path_no_args
+        else:
+            request_path = request_path_no_args + '?' + urlencode(options)
+
+        nonce = str(int(time.time() * 1000))
+        request_url = self.uri + request_path
+
+        self._generate_signature(
+            request_type=verb.upper(),
+            url=request_url,
+            nonce=nonce,
+        )
+
+        self.session.headers.update({
+            'x-api-nonce': nonce,
+        })
+        if data != '':
+            self.session.headers.update({
+                'Content-Type': 'application/json',
+                'Content-Length': str(len(data)),
+            })
+
+        log.debug('Bitcoin.de API Query', verb=verb, request_url=request_url)
+
+        response = getattr(self.session, verb)(request_url, data=data)
+
+        try:
+            json_ret = rlk_jsonloads(response.text)
+        except JSONDecodeError as exc:
+            raise RemoteError('Bitcoin.de returned invalid JSON response') from exc
+
+        if response.status_code not in (200, 401):
+            if isinstance(json_ret, dict) and 'errors' in json_ret:
+                for error in json_ret['errors']:
+                    if error.get('field') == 'X-API-KEY' and error.get('code') == 1:
+                        raise RemoteError('Provided API Key is in invalid Format')
+
+                    if error.get('code') == 3:
+                        raise RemoteError('Provided API Key is invalid')
+
+                raise RemoteError(json_ret['errors'])
+
+            raise RemoteError(
+                'Bitcoin.de api request for {} failed with HTTP status code {}'.format(
+                    response.url,
+                    response.status_code,
+                ),
+            )
+
+        if not isinstance(json_ret, dict):
+            raise RemoteError('Bitcoin.de returned invalid non-dict response')
+
+        return json_ret
+
+    def validate_api_key(self) -> Tuple[bool, str]:
+        "Validates that the Bitcoin.de API key is good for usage in Rotki"
+
+        try:
+            self._api_query('get', 'account')
+            return True, ""
+
+        except RemoteError as e:
+            return False, str(e)
+
+    def query_balances(self, **kwargs: Any) -> Tuple[Optional[Dict[Asset, Dict[str, Any]]], str]:
+        balances = {}
+        resp_info = self._api_query('get', 'account')
```

Comment:
Put this in try/except and catch the RemoteError. Then if there is a RemoteError, catch it, log it and return `None, error_message` just like we do in other exchanges.

Example:

https://github.com/rotki/rotki/blob/4500381d5c527d255256f1a32d8261722ce028c2/rotkehlchen/exchanges/bittrex.py#L306-L315